src/main/java/core/mate/academy/model/ExcavatorProducer.java

```diff
@@ -0,0 +1,14 @@
+package core.mate.academy.model;
+
+import core.mate.academy.service.MachineProducer;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer {
```

Comment:
You are using raw types here. You should always parametrize interfaces and classes if it is possible. This class should implement MachineProducer<Excavator>.